builtin/providers/datadog/resource_datadog_downtime.go

```diff
@@ -0,0 +1,306 @@
+package datadog
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogDowntime() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogDowntimeCreate,
+		Read:   resourceDatadogDowntimeRead,
+		Update: resourceDatadogDowntimeUpdate,
+		Delete: resourceDatadogDowntimeDelete,
+		Exists: resourceDatadogDowntimeExists,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogDowntimeImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"active": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"disabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"end": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"message": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(val interface{}) string {
+					return strings.TrimSpace(val.(string))
+				},
+			},
+			"recurrence": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"period": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"until_date": {
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ConflictsWith: []string{"recurrence.until_occurrences"},
+						},
+						"until_occurrences": {
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ConflictsWith: []string{"recurrence.until_date"},
+						},
+						"week_days": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validateWeekDay,
```

Comment:
Optional, but validateWeekDay be defined here, see this example: https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_kinesis_firehose_delivery_stream.go#L76:

```go
	ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
		value := v.(string)
		if value != "s3" && value != "redshift" && value != "elasticsearch" {
			errors = append(errors, fmt.Errorf(
				"%q must be one of 's3', 'redshift', 'elasticsearch'", k))
		}
		return
	},
```